bbb/oderhs.m

```diff
@@ -4549,14 +4597,16 @@ ccc             call volave(nx, ny, j2, j5, i2, i5, ixp1(0,0), ixm1(0,0),
 
 
 c ... If molecules are present as gas species 2, add ion/atom cooling
-      if(ishymol == 1) then
-        do iy = j2, j5
-          do ix = i2, i5
-            resei(ix,iy) = resei(ix,iy) - vol(ix,iy)*eqpg(ix,iy,2)*
-     .                                     (ti(ix,iy)-tg(ix,iy,2))
-          enddo
-        enddo
-      endif
+      #..zml ion/molecule elastic collisions have been considered in
+      #engbalg subroutine, so comment the following lines...
+#      if(ishymol == 1) then
```

Comment:
I believe that engbalg only accounts for the energy change of molecules due to elastic collisions, without considering the energy change of ions:

      do igsp = 1, ngsp
        do iy = j2, j5
          iy1 = max(0,iy-1)
          do ix = i2, i5
            ix1 = ixm1(ix,iy)
            reseg(ix,iy,igsp)= -( fegx(ix,iy,igsp)-fegx(ix1,iy,  igsp)+
     .                            fegy(ix,iy,igsp)-fegy(ix, iy1,igsp) )
     .                                                + segc(ix,iy,igsp)
            reseg(ix,iy,igsp)= reseg(ix,iy,igsp) + vol(ix,iy)* 
     .                      eqpg(ix,iy,igsp)*(ti(ix,iy)-tg(ix,iy,igsp))+
     .                   cftgdiss(igsp)*psorg(ix,iy,igsp)*tg(ix,iy,igsp)
          enddo
        enddo
      enddo

It seems the energy residuals of the ions due to elastic scattering with molecules is only accounted for in these lines.